File: paleo_workbench/ui/pages/action_header.py
from __future__ import annotations

from PySide6.QtCore import Signal
from PySide6.QtWidgets import (
    QFrame,
    QHBoxLayout,
    QLabel,
    QPushButton,
    QVBoxLayout,
)

from paleo_workbench.ui import tokens


class ActionHeader(QFrame):
    """Top banner of the 成图审核 page — title, action buttons, rules chips."""

    run_requested = Signal()
    export_requested = Signal()
    config_requested = Signal()
    finalize_requested = Signal()

    def __init__(self, parent=None):
        super().__init__(parent)
        self.setObjectName("PanelCard")

        layout = QVBoxLayout(self)
        layout.setContentsMargins(tokens.SPACE_3, tokens.SPACE_3, tokens.SPACE_3, tokens.SPACE_3)
        layout.setSpacing(tokens.SPACE_3)

        self.title_label = QLabel(
            "成图与审核 · — 古地理图（自动质检 + 人工审核）"
        )
        self.title_label.setStyleSheet(
            f"color: {tokens.TEXT_PRIMARY}; font-size: {tokens.FONT_SIZE_TITLE}; font-weight: 600;"
            " border: none; background: transparent;"
        )
        layout.addWidget(self.title_label)

        button_row = QHBoxLayout()
        button_row.setSpacing(tokens.SPACE_2)
        self.run_btn = QPushButton("运行检查")
        self.run_btn.setObjectName("PrimaryButton")
        self.run_btn.setToolTip("运行自动质检规则")
        self.run_btn.clicked.connect(self.run_requested.emit)
        button_row.addWidget(self.run_btn)
        self.config_btn = QPushButton("规则配置")
        self.config_btn.setObjectName("SecondaryButton")
        self.config_btn.setToolTip("当前规则见下方列表（配置面板后续迭代）")
        self.config_btn.clicked.connect(self.config_requested.emit)
        button_row.addWidget(self.config_btn)
        self.export_btn = QPushButton("导出检查报告")
        self.export_btn.setObjectName("PrimaryButton")
        self.export_btn.setToolTip("导出质检报告文件")
        self.export_btn.clicked.connect(self.export_requested.emit)
        button_row.addWidget(self.export_btn)
        self.finalize_btn = QPushButton("专家定稿")
        self.finalize_btn.setObjectName("SecondaryButton")
        self.finalize_btn.setToolTip("将当前古地理图写入 VersionSet 快照并标记为 final")
        self.finalize_btn.clicked.connect(self.finalize_requested.emit)
        button_row.addWidget(self.finalize_btn)
        button_row.addStretch()
        layout.addLayout(button_row)

        self.rules_label = QLabel(
            f"检查规则: {' · '.join(tokens.DEFAULT_QC_RULES)}"
        )
        self.rules_label.setStyleSheet(
            f"color: {tokens.TEXT_SECONDARY}; font-size: 11px;"
            " border: none; background: transparent;"
        )
        layout.addWidget(self.rules_label)

    def update_state(self, reports: list, map_documents: list) -> None:
        horizon = "—"
        if reports:
            linked_id = reports[0].linked_map_document_id
            for doc in map_documents:
                if doc.id == linked_id:
                    horizon = doc.linked_target_horizon or "—"
                    break
        elif map_documents:
            horizon = getattr(map_documents[-1], "linked_target_horizon", None) or "—"
        self.title_label.setText(
            f"成图与审核 · {horizon} 古地理图（自动质检 + 人工审核）"
        )
        self.finalize_btn.setEnabled(bool(map_documents))

        if reports and reports[0].rules:
            chips = " · ".join(reports[0].rules)
        else:
            chips = " · ".join(tokens.DEFAULT_QC_RULES)
        self.rules_label.setText(f"检查规则: {chips}")

        has_maps = bool(map_documents)
        self.run_btn.setEnabled(has_maps)
        self.export_btn.setEnabled(bool(reports))

File: libs/ui_pages_data/src/qt/action_header.cpp
// UI-06 — ActionHeader shell (see qt/action_header.hpp).
#include <pwb/ui_pages_data/qt/action_header.hpp>

#include <QHBoxLayout>
#include <QLabel>
#include <QPushButton>
#include <QVBoxLayout>

#include <pwb/ui_pages_data/qc_summary.hpp>
#include <pwb/ui_pages_data/vocab.hpp>
#include <pwb/ui_shell/style_registry.hpp>

namespace pwb::ui_pages_data::qt {
namespace {

QString pal(const char* token) {
    const auto p = ui_shell::style_palette();
    const auto it = p.find(token);
    return it != p.end() ? QString::fromStdString(it->second) : QString();
}

}  // namespace

ActionHeader::ActionHeader(QWidget* parent) : QFrame(parent) {
    setObjectName(QStringLiteral("PanelCard"));
    auto* layout = new QVBoxLayout(this);
    layout->setContentsMargins(12, 12, 12, 12);  // SPACE_3
    layout->setSpacing(12);                      // SPACE_3

    title_label_ = new QLabel(
        QStringLiteral("成图与审核 · — 古地理图（自动质检 + 人工审核）"),
        this);
    ui_shell::style_bind(title_label_, [] {
        return QStringLiteral(
                   "color: %1; font-size: 14px; font-weight: 600;"
                   " border: none; background: transparent;")
            .arg(pal("TEXT_PRIMARY"));           // FONT_SIZE_TITLE
    });
    layout->addWidget(title_label_);

    auto* button_row = new QHBoxLayout();
    button_row->setSpacing(8);                   // SPACE_2
    run_btn_ = new QPushButton(QStringLiteral("运行检查"), this);
    run_btn_->setObjectName(QStringLiteral("PrimaryButton"));
    run_btn_->setToolTip(QStringLiteral("运行自动质检规则"));
    connect(run_btn_, &QPushButton::clicked, this,
            &ActionHeader::run_requested);
    button_row->addWidget(run_btn_);
    config_btn_ = new QPushButton(QStringLiteral("规则配置"), this);
    config_btn_->setObjectName(QStringLiteral("SecondaryButton"));
    config_btn_->setToolTip(
        QStringLiteral("当前规则见下方列表（配置面板后续迭代）"));
    connect(config_btn_, &QPushButton::clicked, this,
            &ActionHeader::config_requested);
    button_row->addWidget(config_btn_);
    export_btn_ = new QPushButton(QStringLiteral("导出检查报告"), this);
    export_btn_->setObjectName(QStringLiteral("PrimaryButton"));
    export_btn_->setToolTip(QStringLiteral("导出质检报告文件"));
    connect(export_btn_, &QPushButton::clicked, this,
            &ActionHeader::export_requested);
    button_row->addWidget(export_btn_);
    finalize_btn_ = new QPushButton(QStringLiteral("专家定稿"), this);
    finalize_btn_->setObjectName(QStringLiteral("SecondaryButton"));
    finalize_btn_->setToolTip(
        QStringLiteral("将当前古地理图写入 VersionSet 快照并标记为 final"));
    connect(finalize_btn_, &QPushButton::clicked, this,
            &ActionHeader::finalize_requested);
    button_row->addWidget(finalize_btn_);
    button_row->addStretch();
    layout->addLayout(button_row);

    std::string rules;
    for (std::size_t i = 0; i < kDefaultQcRules.size(); ++i) {
        if (i) rules += " · ";
        rules += kDefaultQcRules[i];
    }
    rules_label_ = new QLabel(
        QStringLiteral("检查规则: %1").arg(QString::fromStdString(rules)),
        this);
    ui_shell::style_bind(rules_label_, [] {
        return QStringLiteral(
                   "color: %1; font-size: 11px;"
                   " border: none; background: transparent;")
            .arg(pal("TEXT_SECONDARY"));         // FONT_SIZE_STATUS
    });
    layout->addWidget(rules_label_);
}

void ActionHeader::update_state(const domain::Json& reports,
                                const domain::Json& docs) {
    const ActionHeaderState s = resolve_action_header(reports, docs);
    title_label_->setText(QString::fromStdString(s.title));
    rules_label_->setText(QString::fromStdString(s.rules_line));
    run_btn_->setEnabled(s.run_enabled);
    export_btn_->setEnabled(s.export_enabled);
    finalize_btn_->setEnabled(s.finalize_enabled);
}

}  // namespace pwb::ui_pages_data::qt
